src/mscorlib/Common/System/SR.cs

```diff
@@ -0,0 +1,98 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Resources;
+using System.Runtime.CompilerServices;
+
+namespace System
+{
+    internal static partial class SR
+    {
+        private static ResourceManager s_resourceManager;
+
+        private static ResourceManager ResourceManager
+        {
+            get
+            {
+                if (SR.s_resourceManager == null)
+                {
```

Comment:
Also, there is a code in Environment.cs that protects against infinite recursion when things go really wrong in the runtime/corelib. We need an equivalent here.

@morganbr wrote some code for it in CoreRT, but it is not complete (https://github.com/dotnet/corert/issues/2738).